### DISCUSSION TITLE
SimpleAI.FluxAIO: 强大的Flux控图工具开放公测/ Public beta: Comprehensive control images tools base on Fluxdev

### DISCUSSION THREAD

**Discussion by metercai** [metercai/SimpleSDXL#94] [Announcements]
基于SimpleSDXL2双引擎架构的SimpleAI.FluxAIO最新版已开放公测。本次新版在沿袭Fooocus的快捷操控下，利用内置Comfyd引擎实现了Flux全面控图，并做了非常多的创新，具体看点包括：
1. 预置导航个性化，可排序可收纳。
2. SDXL控图新增姿态控制模块，补齐短板。
3. 全面解锁Flux多种控图：风格迁移、边缘检测、结构深度、脸部交换、身体姿态，出图把控得心应手。
4. Flux版的加细节放大、高清修复和变化演绎。分块放大可持续多轮放大提升分辨率到32k，细节满满。
5. Flux版的内部重绘和扩图，修图质量再上新台阶。
6. 前端完整的多用户支持，可远程绑定不同用户，且出图日志和相册空间，相互隔离，互不干扰。
7. 管理员用户可一键进入Comfyd操控界面，拉线接线任你驰骋。
8. 基础环境（Pytorch2.4.1+cuda12.4）全面升级，出图速度提升10%以上。

SimpleAI based on SimpleSDXL2 dual engine architecture The latest version of FluxAIO has been open for public testing. This new version, following the UI of Fooocus, utilizes the built-in Comfyd engine to achieve comprehensive control images base on Fluxdev and has made many innovations, including:

1. Preset personalized navigation, which can be sorted and stored.
2. Add pose control module to the SDXL control chart and fill in the gaps.
3. Fully unlock multiple control images for Flux: imageprompt, canny, depth, faceswap, openpose, making it easy to control the images process.
4. The Flux version features: upscale, high-definition restoration, and vary. Segmented upscale can sustain multiple rounds of upscale to increase resolution to 32k, with full details.
5. The inpaint and outpaint of the Flux version have taken the quality of the retouching to a new level.
6. The front-end has complete multi-user support, which allows remote binding of different users, and the image log and album space are isolated and do not interfere with each other.
7. Administrator users can easily access the Comfyd control interface with just one click, allowing you to freely connect wires to any nodes for yourself workflow.
8. The basic environment (Pytorch 2.4.1+CUDA 12.4) has been fully upgraded, resulting in an increase of over 10% in image processing speed.

由于本次加入身份系统的基础设施，需要提醒注意的是：
1. 开启软件后，系统默认为游客身份，即自动具有guest身份标识，出图保存在users/guest下。游客功能与Fooocus原版保持一致，新增Comfyd引擎系列功能只能在绑定身份后开启。未来的新增功能也需要身份才能解锁。
2. 按照系统提示绑定身份即可解锁所有功能。并且第一个绑定身份默认为管理员，具有超级管理权限，出图保存在users/admin_开头的目录内。
3. 远程绑定的普通用户出图保存在users的特殊目录内，不易在本地进行浏览。
4. 不要轻易移动系统根目录，会导致身份系统重建。
5. 身份绑定成功后第一时间抄写身份口令保存在安全地方，丢失无法找回。第二时间要导出身份二维码，以支持离线身份漫游或重构系统的再次绑定。
6. 身份的唯一性来自昵称+手机号的组合，同一手机号限定最多3个昵称。请慎重选择昵称，坚持长期使用。

Due to the infrastructure of the identity system being added this time, it should be noted that:
1. After starting, the system defaults to guest identity, which automatically has a guest identity identifier, and the image is saved under users/guest. The tourist function remains the same as the original version of Fooocus, and the newly added Comfyd engine series function can only be enabled after binding the identity. Future new features also require identity to unlock.
2. Follow the system prompts to bind your identity to unlock all functions. And the first bound identity defaults to administrator, with super management privileges, and the output is saved in the directory starting with users/admin_.
3. The images of ordinary users bound remotely are saved in the special directory of users, which is not easy to browse locally.
4. Do not easily move the system root directory, as it may cause identity system reconstruction.
5. After successful identity binding, copy the identity password and save it in a secure place as soon as possible. If lost, it cannot be retrieved. The second time is to export the identity QR code to support offline identity roaming or re binding of the reconstruction system.
6. The uniqueness of identity comes from the combination of nickname and phone number, with a maximum of 3 nicknames for the same phone number. Please choose your nickname carefully and stick to long-term use.

下载安装方式和以前的版本一样，地址为：
1. 基础模型包：https://huggingface.co/metercai/SimpleSDXL2/resolve/main/models_base_simpleai_1214.zip
2. FluxAIO模型包：https://huggingface.co/metercai/SimpleSDXL2/resolve/main/models_flux_aio_simpleai_1214.zip
3. 程序包： https://huggingface.co/metercai/SimpleSDXL2/resolve/main/SimpleAI.FluxAIO_dev.exe.7z1215

The download and installation method is the same as the previous version, and the address is:
1.  Basic model package: https://huggingface.co/metercai/SimpleSDXL2/resolve/main/models_base_simpleai_1214.zip 
2. FluxAIO model package: https://huggingface.co/metercai/SimpleSDXL2/resolve/main/models_flux_aio_simpleai_1214.zip
3. Program Package: https://huggingface.co/metercai/SimpleSDXL2/resolve/main/SimpleAI.FluxAIO_dev.exe.7z1215

**Comment by DavidDragonsage** [comment]
Well this is exciting! You've been busy! 🙂👍

**Comment by DavidDragonsage** [comment]
I had no problems with the installation. The two archives to update the SimpleModels folder installed over top of the old SimpleSDXL2 SimpleModels folder without affecting the operation of SimpleSDXL2, so that was good.

The problem came when I wanted to acquire an administrative identity. The first problem is that the Identity Centre information should be available in English. I made an online translation but it was still not obvious what to do. I eventually realized that the title "IdentityCenter" was actually a button that you needed to click.

So I entered a name and number but I have no idea what to type in the "Verify Identity" box. The name and/or number did not work.

There was a step that called for a QrCode that I thought was probably optional. I have no idea what QrCode I would put in there. Lots of mysteries with this procedure!

I could see where this whole security feature would be useful in a multi-user environment like a school or college but I do not see what benefit it would be for the average user. My house is secure and my computer is secure - any further layers of security have no value to me.

**Comment by DavidDragonsage** [comment]
OK, so I entered a QR Code, the URL of the Facebook Pure Fooocus group, https://www.facebook.com/groups/fooocus, and noticed that SimpleAI wanted to "call home", i.e. access the Internet. Perhaps foolishly, I actually allowed it to do so. In any event no verification code appeared in the box.

I do not know what is going on here, but I am not impressed! The founders of Fooocus made sure that Fooocus did not need to connect to the Internet while in use, in part so that users with controlled Internet access, such as in China, were always able to use the program. 

Perhaps there is some simple explanation why this branch of Fooocus now needs to do so, but from what I know at this point it seems suspicious and unethical to me that this open source program is now partially locked unless one is prepared to surrender one's identity.

And if this is the new policy for SimpleAI, it needs to be very clearly stated. When I read about new security in the announcement, I assumed that it was at the local level only. Why would it be otherwise? This level of global security should be on an "opt-in" basis, meaning that it only should be available by user request. I am only creating images, not doing online banking!

There is one last thing, of course I entered a fake phone number. I am certainly not giving out my phone number to random people on the Internet. But that would not matter anyway because _I do not have a cell phone_ - I use landline only (believe it or not, the whole world does not have cellphones). So if the security code was going to be sent as a text, that is obviously not going to work.

**Comment by DavidDragonsage** [comment]
Well that is interesting! I cannot even run the basic Flux features that were available in the latest versions of SimpleSDXL2. Unless these security problems can be resolved, SimpleAI is essentially useless to me. I will be letting the folks at Pure Fooocus know not to bother upgrading at this point.

**Comment by metercai** [comment]
The addition of identity infrastructure in this version is to implement the subsequent community sharing plan: P2P based model sharing and computing power sharing. Fluxdev's demand for computing power has significantly increased, and through community sharing programs, it hopes to eliminate dependence on large computing power.

SimpleAI.FluxAIO will not only be an application for personal use, but also a node that can be serviced.

**Reply by DavidDragonsage** [reply]
Participation in such a program and the security protocols required to implement such a program should be completely optional. 

Effectively locking out users such as myself from "personal use" is not going to help the goal of "community sharing" - quite the opposite! You need to build up mutual respect and trust if you hope to establish a community. This is not the way to do it.

I appreciate you sharing your actual goals with this project. Considering the reputation that the government of China has for security exploits in the west, I doubt you will have many participants from Europe or the Americas.

On a personal note, in part of my youth I grew up in a Chinese community on the west coast of Canada. I greatly appreciated the order, purpose, kindness and sanity that these young friends brought to my life in an otherwise difficult period. I have spent my life fighting racial prejudice and hatred of those who are different. In my retirement I continue to do so at Pure Fooocus and I intentionally celebrate the beauty of ethnic diversity in all my artistic projects.

I appreciate all the work you have put into this open source project, and your goal of building community in order to empower more people to create works of art is quite wonderful! That is also a primary motivation for me in promoting Fooocus and its branches.

I do hope you will somewhat modify your approach. I will continue to monitor this thread. 🙂

**Reply by metercai** [reply]
The newly added identity infrastructure is not an account system like Facebook, but a distributed system where users control private keys, similar to the private keys in blockchain systems. The situation where Facebook blocks President Trump's account is completely impossible to happen. Just added one-time authentication for creating private keys to prevent abuse. We do not want a large number of robots to obtain private keys and participate in community sharing programs, which could lead to the proliferation of junk information. Daily use only requires local authorization of the private key without network involvement.


This is our Chinese explanation of the identity system on the interface:
系统遵循分布式身份管理机制，即用户自主掌控身份私钥，授权AI节点使用身份；AI节点私有部署，管理多用户相互隔离的数字空间；社区节点保存身份加密副本用于追溯和自证。在多方协作下共同保障隐私安全、身份可信及跨节点互认。以此构建"和而不同"的开源社区生态。
The system follows a distributed identity management mechanism, where users independently control their identity private keys and authorize AI nodes to use their identities; Private deployment of AI nodes to manage digital spaces where multiple users are isolated from each other; Community nodes store encrypted copies of their identities for traceability and self verification. Through multi-party collaboration, we jointly ensure privacy and security, trustworthy identity, and cross node mutual recognition. Build an open source community ecosystem that is harmonious but diverse.

In addition, our community sharing program does not involve text or discussion, and we do not want to incorporate ideology and various biases. It only involves model sharing and computing power sharing to lower the threshold for use. You are not familiar with the real situation in China.

**Comment by DavidDragonsage** [comment]
This still leaves me out in the cold because I do not have a cellphone. It is not that unusual. My forty year old son who lives in another city is highly technological but for several years chose to have neither a landline nor a cellphone. My daughter who lives in the USA only uses her cellphone for travelling. We all communicate with each other through the Internet.

I understand your concerns about robots, that make perfect sense. And really I am not familiar with any country where I have not lived, which is my loss. But those who do not obtain keys should not be permitted to enter into community sharing. I am familiar with key control, I used to work for the government of Canada as an electronics and informatics (computer) technologist.

What I do not understand is why the security system cannot be turned off for local use only. Well in effect it is turned off in the case of SimpleAI, but for some reason that blocks out the use of Flux and the other platforms that were available in SimpleSDXL2. That does not make sense to me.

BTW, when I referred to the "reputation that the government of China has for security exploits in the west", personally I do not trust my own government concerning security exploits either. Having been responsible for security in a portion of that government, I am very aware that I need to protect my own personal security. My wife, who is an American Canadian, is even more sensitive to that threat - with good reason.

**Reply by metercai** [reply]
In China, mobile phones have become very common, and mobile phones and mobile payments have become necessities of life. From the elderly to the students, most of them have mobile phones. It is very common and effective to use a mobile phone verification code to confirm identity. Perhaps users in different regions have different habits, so how to cater to a wider audience is a matter that needs to be considered in the future. For example, can we avoid robots by confirming via email?

Identity is the infrastructure of a community, not an accessory. In addition to native use, SimpleSDXL also supports multi-user access within the LAN and Internet access after joining the vpn tunnel. That is to say, the SimpleSDXL service at home pc can be accessed from anywhere with a mobile phone. So, SimpleSDXL needs a unique identity system. This is also the innovation of SimpleSDXL.

As stated on the project homepage, “Enhanced version of Fooocus for SDXL, more suitable for Chinese and Cloud”. SimpleSDXL is not just a branch of Fooocus. If you have any good suggestions, you can honestly put them forward. Make SimpleSDXL better and more widely used.

**Comment by DavidDragonsage** [comment]
Well I have followed up on my assignment! The exchange of a token through a mobile device does seem to be the most practical and lowest cost method of authentication. As a non-mobile using family, the most common alternative available for me is automated voice callback, from which I enter a one-time authentication code. In the case of the provincial medical service they actually issue a physical token device that produces codes - a ridiculously clunky solution! Both of these methods involve a costly system that would not be practical for you.

Many corporate services use email callback as an alternative. They send you an email, then you simply confirm the email or enter a code online. This usually ties in to credit card use which of course locks out robots, so again not practical for SimpleAI.

Then there is "CAPTCHA" with a free Python plugin available: http://www.captcha.net/  That could be a possibility but I think you would need to operate a website, so again there are costs involved (although there are websites that are more or less free, paid for by advertising). Actually you can operate CAPTCHA on the localhost but I do not know if that would be effective.

As an aside, I was looking at household mobile usage in North America. 98% in the USA, but 94% in Canada. There is starting to be a movement towards dumbphones, which do only text and voice - no Internet. Service providers are resisting this because they would make less money from it. There is a Lightphone that does not do text either, but it is not available in Canada yet. This movement reflects a growing rebellion against over-connectedness, online addiction, corporate and government control, physical tracking and general invasion of privacy.

But I do have an idea rolling around in my head - I do not know if it is practical but I will share it anyway (we call it "spitballing"). Could there be a way of making each download somehow unique? So that only the downloader could use that particular copy? Perhaps that does mean logging on to another site besides Github, at least for that one little file that creates the uniqueness. What I mean is, I think it would an unlikely exploit for the bad guys to run multiple copies of the 9GB SimpleAI application so they could abuse community sharing. In effect you could create a token system by doing so.

However, security is a rabbit hole! You could easily spend more time and effort on security than on your actual application!

But the basic question is, are you trying to make usage of community sharing compulsory? Because otherwise it should be "relatively easy" (I am enough of a programmer to say that there is no such thing!) to compartmentalize the code so that all features are available locally, but if the user wants to participate in community sharing then of course they need to be authenticated.

**Comment by metercai** [comment]
The latest public beta version has opened up access for guest to use AIO presets , including FluxAIO, kolorsAIO, and SD15AIO presets. However, the personalization of preset navigation and MiniCPM multimodal describe the image to prompt still require activating the identity system. You can give it a try and experience the outstanding Fooocus handling of Flux, Kolors, and SD1.5 models

**Comment by DavidDragonsage** [comment]
@metercai Thank you for unlocking almost all the new features of SimpleAI for those who are unable to or do not wish to participate in the new security system. I am sure that a lot of our users at the Facebook Pure Fooocus group will appreciate that. One of members reported on the new release already and was quite happy. I just had a chance to try it out today and it is looking good! I see that the only things that I cannot do without a security key are some Comfy manipulations, but that I am not really interested in that.

I did find another security system that does not need a mobile phone, It is the Authenticator Extension that Github has just implemented. You have probably used it already to continue developer access privileges, as I have done also.
https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/configuring-two-factor-authentication
For Firefox I used this add-on:
https://addons.mozilla.org/en-US/firefox/addon/auth-helper/?src=external-github

In the meantime I have opened a new fork of SimpleSDXL2, based on the 0916 release. While I really like SimpleSDXL2 - it is a treasure to use - I wanted to bring it back a little closer to mainline Fooocus.

Of course, my primary concern was the lack of a dropdown for preset selection, which in practice means that I have to limit SimpleSDXL2 presets to non-SDXL use only, using the Topbar system. For that reason alone I am still using mainline Fooocus for all SDXL work, although there are also one or two other advanced feature such as "Mixing Image Prompt and Vary/Upscale" and "Mixing Image Prompt and Inpaint" that are also missing from SimpleSDXL2.

As much as possible I am maintaining convenience features for Chinese users, for example the Topbar presets will default to being active if lang=="cn", and to Dropdown presets otherwise - but that feature will also be switchable within the UI as well as within config.txt.

Anyway, we are all dealing in open source so if you see anything you might like of course I encourage you to pick it up once it goes live! The new repo is called FooocusPlus and it is still in pre-release status, here:
https://github.com/DavidDragonsage/FooocusPlus/tree/FooocusPlus

**Reply by iwr-redmond** [reply]
If you can please activate the Issues and/or Discussions features in your new repository, @DavidDragonsage, I would be happy to make some suggestions about removing simpleai_base code that will be unnecessary for your purposes (per [issue 1](https://github.com/metercai/simpleai_base/issues/1)).

**Reply by DavidDragonsage** [reply]
@iwr-redmond  Thank you, that will be great! Yes, I have no interest in making FooocusPlus multi-user, networkable or community resource sharing - quite the opposite!

I have opened the Discussions and Issues sections at FooocusPlus:
https://github.com/DavidDragonsage/FooocusPlus/discussions/
